### PR TITLE
Create ADR for Versioning

### LINKED ISSUE #1: where to place version identifier for json objects
Option 1:
Place version in top level json object:
```
{
  "version": "3.0.0",
  "identifier": "11483f23-95bf-424a-98a5-ee5868c85c3e",
  "metadata": { ... },
...
}
```

Option 2:
Use an additional json object as wrapper for the json object.
The wrapper object indicate the type and version of the wrappen object:
```
{
  "version": "3.0.0",
  "ComputationTemplate": {
    "identifier": "11483f23-95bf-424a-98a5-ee5868c85c3e",
    "metadata": {...},
    ...
  }
}
```

Option 2 makes it easy to add information that is only required when the json is transferred to other applications or stored. The domain object itself is not modified, but stored and transferred with additional metadata, like an envelope.

### PR DESCRIPTION
**Pull request by Legion2**
close #1

### CODE REVIEW

File: docs/viplab3.0/versioning.md [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+rules apply when a 3.1.x client talks to a 3.0.x api. The api will process the message according to the 3.0.x rules and extra information might be discarded.
+
+The client has to determine the correct endpoint versions to use. When the client will use a ComputationTemplate with version 3.x.x the corresponding
+websocket api will be ws://example.com/v3/compuation. All other messages used in any further communication between the websocket-api and the client
+are also expected to be version 3.x.x. There is no possiblity to mix different versions. If a client wants to use a v4 api, he has to reconnect to the correct
+endpoint at ws://example.com/v4/compuation
+
+The objects have been devided into two groups:
+- Long living object: These objects are expected to be stored by clients. These objects are ComputationTemplate, the ComputationTask and the ComputationResult
+- Action Wrappers: Computation and Result
```
Comment (Legion2) [inline docs/viplab3.0/versioning.md]: [Create Computation](https://github.com/VirtualProgrammingLab/viplab-websocket-api/blob/master/schema/create-computation-schema.json) and [Computation](https://github.com/VirtualProgrammingLab/viplab-websocket-api/blob/master/schema/computation-schema.json) are short lived interaction objects/wrappers. They only exist for the purpose of communicating information in the websocket connection and as mentioned are versionen by the API endpoint.

But Result as it is currently discussed in VirtualProgrammingLab/viplab-websocket-api#14 is not only an object which is created for the websocket API. It is created by the backends and send via the websocket API to the clients, so it must have a version and can not be versioned with the endpoint.

Comment (pascalseeland) [reply docs/viplab3.0/versioning.md]: For Result, I think we have a naming issue. The concept described in  VirtualProgrammingLab/viplab-websocket-api#14 does belong to the ComputationResult. And those would, as being long living objects, have an version.
The Result I mentioned in the Action Wrappers is actually the message type that gets send to the client from the api.
Maybe, we need to define three types of objects. Long living (they get versioned), message types(unversioned, like Create Computation and Result) and Wrappers (unversioned, Computation). The unversioned types are only specified at an endpoint level, and require the endpoint version.

Comment (Legion2) [reply docs/viplab3.0/versioning.md]: Ok, yes

**Comment by Legion2** [comment]
LGTM. I can not approve the PR because I created it.

**Pull request merged**